hydra_agent/agent.py

```diff
@@ -140,5 +153,58 @@ def process_embedded(self, embedded_resources: list) -> None:
                 embedded_resource['parent_type'],
                 embedded_resource['embedded_url'])
 
+    # Below are the functions that are reponsible to process Socket Events
+    def on_connect(self) -> None:
+        """Method executed when the Agent is successfuly connected to the Server
+        """
+        logger.info('Socket Connection Established - Synchronization ON')
+        self.modifications_table = super().get(self.entrypoint_url_temp +
+                                              '/modification-table-diff').json()
+        if self.modifications_table:
+            self.last_job_id = self.modifications_table[0]['job_id']
+
+    def on_disconnect(self):
+        """Method executed when the Agent is disconnected
+        """
+        pass
+
+    def on_update(self, data) -> None:
+        """Method executed when the Agent receives an event named 'update'
+        This is sent to all clients connected the server under the designed Namespace
+        """
+        if data['last_job_id'] == self.last_job_id:
+            new_rows = [data]
```

Comment:
Won't this always be one element? (See #406 in hydrus, L244 helpers.py)

Reply:
Yeah it will, I did it like this to transform from a dict to a list of length 1 so the code below remains the same and can be used for both scenarios- fetching multiple updates or just this one from the socket.